### ISSUE TITLE
Add a localisation directory to get translations to happen

### ISSUE THREAD

**Issue by Pomax** [Pomax/BezierInfo-2#51]
Starting with `cn`

**Comment by mashabow** [comment]
I would like to start on Japanese localization; where should I put translated texts? In what format?

**Comment by Pomax** [comment]
I'm still looking at good solutions, but it's all going to end up involving "keyed text" somehow so it's probably easiest to just start with copying the section files and just translating inside of those. Stripping the markup around it and putting it in the correct format (prop file, json, whatever ends up the most usable) is easier than going the other way around.

**Comment by mashabow** [comment]
> so it's probably easiest to just start with copying the section files and just translating inside of those.

Thank you. Now I have started in this way, see https://github.com/mashabow/BezierInfo-2/commit/d7973eb3fd031dd345157e3e509f1f279c52b20f

**Comment by Pomax** [comment]
cool, I'm trying to set up [Pontoon](https://pontoon.mozilla.org) asap so that you can use in-page tools for translating instead, and then once that works it should be fairly easy to start moving these translations into the `messages.properties` file format Pontoon uses.

**Comment by Pomax** [comment]
Pontoon was a bust. However, I've set up an easier translation system involving content as markdown.

The following sections currently show this off:

- https://github.com/Pomax/BezierInfo-2/tree/master/components/sections/preface
- https://github.com/Pomax/BezierInfo-2/tree/master/components/sections/whatis
- https://github.com/Pomax/BezierInfo-2/tree/master/components/sections/explanation

Each of these directories have a `content.en-GB.md` that can be copied as a file `content.cn-ZH.md` or `content.jp-JP.md`, and can then be translated without all the JSX etc. getting in the way.

I will be converting all the other sections this way over the next few days as well, so that localizing is simply a matter of translating sections as natural, running text in their own file, without any weirdness around translating "per paragraph" or even "per string".

**Comment by thrillerist** [comment]
Good job, I'll add some chinese sections later.

**Comment by mashabow** [comment]
Sounds good, I will move on later.

> `content.cn-ZH.md` or `content.jp-JP.md`

They should be `content.zh-CN.md` and `content.ja-JP.md` respectively, since language tags are basically defined as `language-REGION`. `content.en-GB.md` is OK.

**Comment by Pomax** [comment]
it's a bit free form because the make-localizations.js is going to basically look for "whatever code you give it" (once that's been written into it) but you're right, using correct locale-language codes is wise.

**Comment by Pomax** [comment]
Sections 1-10 have been converted to markdown-based content, and the main bezierinfo article has been updated.

**Comment by Pomax** [comment]
@thrillerist supplied a Chinese localized preface, which is now visible on http://pomax.github.io/BezierInfo-2/?locale=zh-CN

**Comment by thrillerist** [comment]
@Pomax Great!

**Comment by Pomax** [comment]
updated the initial comment with a link to a wiki page that details how to localize. It'll also be where general and specific locale policies can be found (right now it's pretty simple =)

**Comment by Pomax** [comment]
@mashabow supplied a Japanese localized sections 1 and 2, which are now visible on http://pomax.github.io/BezierInfo-2/?locale=ja-JP

**Comment by mashabow** [comment]
@Pomax Thanks!

**Comment by Pomax** [comment]
merged in sections 1 and 2 for `zh-CN`, and localized sections 11-20 for `en-GB`

**Comment by Pomax** [comment]
merged in the preface for `ja-JP`

**Comment by Pomax** [comment]
merged section 3 for `zh-CN`

**Comment by Pomax** [comment]
merged section  3 for `ja-JP`

**Comment by Pomax** [comment]
converted sections 21-30 for localization, and also completely rewrote the LaTeX conversion so as to allow localized LaTeX as well. The new system relies on an actual LaTeX engine (XeLaTeX in my case, because it's a native unicode TeX implementation with excellent OpenType font support) and can deal with Chinese or Japanese in LaTeX text and math settings perfectly fine.

**Comment by Pomax** [comment]
merged section 4 for `zh-CN`, regenerated the content with open source fonts for all locales (though English can do with a slightly thinner font to be honest)

**Comment by Pomax** [comment]
I've switch the local handling from query parameter to actual url, so the versions are now:

- English: http://pomax.github.io/BezierInfo-2/ and http://pomax.github.io/BezierInfo-2/en-GB/
- Japanese: http://pomax.github.io/BezierInfo-2/ja-JP/
- Chinese: http://pomax.github.io/BezierInfo-2/zh-CN/

I've update the language switcher accordingly, and hopefully I caught all the links in the content (relative content like images and stylesheets etc. needed a `../` prefix so they'd resolve properly)

**Comment by Pomax** [comment]
merged in section 5 for  `zh-CN`

**Comment by Pomax** [comment]
merged section 4 for `ja-JP`

**Comment by Pomax** [comment]
merged in section 5 for `ja-JP`

**Comment by Pomax** [comment]
merged section 6 for `zh-CN`

**Comment by Pomax** [comment]
merged section 6 for `ja-JP`

**Comment by Pomax** [comment]
Incidentally, I wrote a blog post on what's going on with this from a technical standpoint, disguised as a blog post on Localization. Maybe it's the other way around... anyway, link: http://pomax.github.io/1489108158510/localization-is-hard

**Comment by Pomax** [comment]
merged section 7 for `ja-JP`

**Comment by Pomax** [comment]
merged section 7 for `zh-CN`, and finished all sections for `en-GB`

**Comment by Pomax** [comment]
merged section 8 for `ja-JP`

**Comment by Pomax** [comment]
merged section 8 for `zh-CN`

**Comment by Pomax** [comment]
merged section 9 for `ja-JP`

**Comment by Pomax** [comment]
merged section 9 for `zh-CN`

**Comment by Hameck** [comment]
Hi, I'm preparing a pt-BR section (Brazilian portuguese), in order to merge do I need to finish translating all sections or can I do so by parts?

**Comment by Pomax** [comment]
Nope, the other languages aren't full localizations either, feel free to open a PR for however much you wanted to do, or even individual PRs for each section you've localized
